tests/components/matter/test_update.py

```diff
@@ -169,3 +194,185 @@ async def test_update_install(
     state = hass.states.get("update.mock_dimmable_light")
     assert state.state == STATE_OFF
     assert state.attributes.get("installed_version") == "v2.0"
+
+
+async def test_update_install_failure(
+    hass: HomeAssistant,
+    matter_client: MagicMock,
+    check_node_update: AsyncMock,
+    update_node: AsyncMock,
+    updateable_node: MatterNode,
+) -> None:
+    """Test update entity service call errors."""
+    state = hass.states.get("update.mock_dimmable_light")
+    assert state
+    assert state.state == STATE_OFF
+    assert state.attributes.get("installed_version") == "v1.0"
+
+    await async_setup_component(hass, "homeassistant", {})
+
+    check_node_update.return_value = MatterSoftwareVersion(
+        vid=65521,
+        pid=32768,
+        software_version=2,
+        software_version_string="v2.0",
+        firmware_information="",
+        min_applicable_software_version=0,
+        max_applicable_software_version=1,
+        release_notes_url="http://home-assistant.io/non-existing-product",
+        update_source=UpdateSource.LOCAL,
+    )
+
+    await hass.services.async_call(
+        "homeassistant",
+        "update_entity",
```

Comment:
Otherwise, please use constants
```suggestion
        HA_DOMAIN,
        SERVICE_UPDATE_ENTITY,
```